core/client/hdfs/src/main/java/alluxio/hadoop/AlluxioFileSystem.java

```diff
@@ -54,4 +57,30 @@
   public int getUriDefaultPort() {
     return Integer.parseInt(PropertyKey.MASTER_RPC_PORT.getDefaultValue());
   }
+
+  @Override
+  public void checkPath(Path path) {
+    URI uri = path.toUri();
+    String thatScheme = uri.getScheme();
+    String thatAuthority = uri.getAuthority();
+    if (thatScheme == null) {
+      if (thatAuthority == null) {
+        if (path.isUriPathAbsolute()) {
+          return;
+        }
+        throw new InvalidPathException("relative paths not allowed:" + path);
+      } else {
+        throw new InvalidPathException("path without scheme with non-null authority:" + path);
+      }
+    }
+
+    if (!StringUtils.equals(this.getUri().getHost(), uri.getHost())) {
+      throw new InvalidPathException(
+          "path not have same host: " + path + ", expected: " + this.getUri());
+    }
```

Comment:
I don't understand - compared with the default implementation https://github.com/apache/hadoop/blob/d7c5400fbcd31ccb8d2301e244fcd7969fd0b77e/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/AbstractFileSystem.java#L390-L396

your change seems to just ignore the comparison of schemes.

If this change allows your tests to pass, then it must be that the scheme strings are not equal in #14648. However, from the exception message they look the same.

Can you please try to do a live debugging to see what are the schemes exactly at runtime?

Reply:
My implementation here is trying to modify the judgment for ports instead of schemes.As for the different logs you see, it is because of the different version of `AbstractFileSystem`.